cypress/e2e/tests/migration/applicationinventory/analysis/downloaded_report.test.ts

```diff
@@ -0,0 +1,159 @@
+*/
+/// <reference types="cypress" />
+
+import {
+    cleanupDownloads,
+    deleteAllMigrationWaves,
+    deleteApplicationTableRows,
+    getRandomAnalysisData,
+    getRandomApplicationData,
+    login,
+    validateTextPresence,
+} from "../../../../../utils/utils";
+import { GeneralConfig } from "../../../../models/administration/general/generalConfig";
+import { Analysis } from "../../../../models/migration/applicationinventory/analysis";
+import { AnalysisStatuses, ReportTypeSelectors } from "../../../../types/constants";
+import { singleApplicationColumns } from "../../../../views/issue.view";
+import { dependencies, issues, technologies } from "../../../../views/common.view";
+
+/**
+ *
+ * This test is divided into three suites due to its specific needs.
+ * The test must access an external URL, download a report, extract it, and open it locally for further testing.
+ *
+ * Cypress cannot handle requests to two different hosts within the same test unless using the cy.origin command. (see https://docs.cypress.io/api/commands/origin)
+ * However, this command does not allow local files as an origin. (see https://github.com/cypress-io/cypress/issues/26670)
+ *
+ * Therefore, the only way to dynamically conduct the test is to split it into three parts:
+ * 1. Analyze an app, download, and extract the report
+ * 2. Conduct tests on the local interface of the report
+ * 3. Delete the test data.
+ *
+ * For this to work, the app's name must be static, as it needs to be shared across all three suites, and random variables would reset after each describe.
+ */
+
+const appName = "Downloaded-Report-Test-App";
+
+describe(["@tier2"], "Prepare Downloaded Report", function () {
+    it("Download and extract report", function () {
+        cy.fixture("application").then(function (appData) {
+            cy.fixture("analysis").then(function (analysisData) {
+                login();
+                cleanupDownloads();
+                deleteAllMigrationWaves();
+                deleteApplicationTableRows();
+                GeneralConfig.enableDownloadReport();
+                const app = new Analysis(
+                    getRandomApplicationData("SourceApp", {
+                        sourceData: appData["bookserver-app"],
+                    }),
+                    getRandomAnalysisData(analysisData["source_analysis_on_bookserverapp"])
+                );
+                app.name = appName;
+                app.create();
+                app.analyze();
+                app.verifyAnalysisStatus(AnalysisStatuses.completed);
+                app.downloadReport(ReportTypeSelectors.HTML);
+                cy.task("unzip", {
+                    path: "cypress/downloads/",
+                    file: `analysis-report-app-${app.name}.tar`,
+                });
+                cy.verifyDownload(`analysis-report-app-${app.name}/index.html`);
+            });
+        });
+    });
+});
+
+describe(["@tier2"], "Test Downloaded Report UI", function () {
```

Comment:
Let's rename this "test static report UI"

Reply:
Solved!

Reply:
I meant the file name itself.